sway/desktop/xwayland.c

```diff
@@ -14,6 +14,33 @@
 #include "sway/input/input-manager.h"
 #include "log.h"
 
+static void unmanaged_handle_destroy(struct wl_listener *listener, void *data) {
+	struct sway_xwayland_unmanaged *sway_surface =
+		wl_container_of(listener, sway_surface, destroy);
+	wl_list_remove(&sway_surface->destroy.link);
+	wl_list_remove(&sway_surface->link);
+	free(sway_surface);
+}
+
+static void create_unmanaged(struct wlr_xwayland_surface *xsurface) {
+	struct sway_xwayland_unmanaged *sway_surface =
+		calloc(1, sizeof(struct sway_xwayland_unmanaged));
+	if (!sway_assert(sway_surface, "Failed to allocate surface")) {
+		return;
+	}
+
+	sway_surface->wlr_xwayland_surface = xsurface;
+
+	wl_signal_add(&xsurface->events.destroy, &sway_surface->destroy);
+	sway_surface->destroy.notify = unmanaged_handle_destroy;
+
+	wl_list_insert(&root_container.sway_root->xwayland_unmanaged,
+		&sway_surface->link);
+
+	// TODO: damage tracking
```

Comment:
Do we need to at least damage the output whole here for this to work?

Reply:
Actually we don't (yet) track damage for popups, subsurfaces and unmanaged views (which are all the same kind of surface). It works because we're lucky. Will definitely fix that in an upcoming PR.